File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;

class Gigasecond {

    LocalDateTime birth;
    static long GIGASECOND = 1_000_000_000L;

    Gigasecond(LocalDate birthDate) {
        birth = birthDate.atStartOfDay();
    }

    Gigasecond(LocalDateTime birthDateTime) {
        birth = birthDateTime;
    }

    LocalDateTime getDateTime() {
        return birth.plusSeconds(GIGASECOND);
    }
}

File: kotlin/gigasecond/src/main/kotlin/Gigasecond.kt
import java.time.LocalDate
import java.time.LocalDateTime

class Gigasecond(val startTime: LocalDateTime) {
    private val gigasecond = 1_000_000_000L

    // convert a LocalDate argument to LocalDateTime
    constructor(startDate: LocalDate) : this(startDate.atStartOfDay())

    val date: LocalDateTime = startTime.plusSeconds(gigasecond)
}
